.github/workflows/test-build-deploy.yml

```diff
@@ -0,0 +1,117 @@
+name: ci
+on:
+  push:
+    branches: [master]
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+** # Tag filters not as strict due to different regex system on Github Actions
+  pull_request:
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/cortexproject/build-image:update-golang-1.14.9-eb0c8d4d2
```

Comment:
Do you think we can use something similar to `defaults: &defaults` in the CircleCI yaml, in order to avoid the repetition of `image: quay.io/cortexproject/build-image:update-golang-1.14.9-eb0c8d4d2` and being able to define the build image only once in this file?

Reply:
There exists a "defaults" field for GitHub actions, but it automatically applies to all jobs in the workflow. The integration and integration-configs-db jobs that will be added with PR 2/3 of the migration do not use the default container so using a default would require explicitly removing the default container from those jobs by setting them to None.

We would like to reuse setup commands and default containers with something like commands in CircleCI, but this feature is not available yet on GitHub Actions.

Status: https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851

Reply:
GHA doesn't support yaml anchors yet because of some security reasons but they are actively working on it: https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851/14

Reply:
Would it be possible to use env variable instead?

Reply:
_This is not a block for getting this PR merged. We can discuss it separately._